RFS/src/main/java/org/opensearch/migrations/bulkload/worker/IndexRunner.java

```diff
@@ -33,44 +34,56 @@ public IndexMetadataResults migrateIndices(MigrationMode mode, ICreateIndexConte
         repoDataProvider.getIndicesInSnapshot(snapshotName)
             .stream()
             .forEach(index -> {
-                CreationResult creationResult;
+                List<CreationResult> creationResults;
                 if (skipCreation.test(index.getName())) {
                     log.atInfo()
                         .setMessage("Index {} was not part of the allowlist and will not be migrated.")
                         .addArgument(index.getName())
                         .log();
-                    creationResult = CreationResult.builder()
+                    creationResults = List.of(CreationResult.builder()
                         .name(index.getName())
                         .failureType(CreationFailureType.SKIPPED_DUE_TO_FILTER)
-                        .build();
+                        .build());
                 } else {
-                    creationResult = createIndex(index.getName(), mode, context);
+                    creationResults = createIndex(index.getName(), mode, context);
                 }
 
-                results.index(creationResult);
+                creationResults.forEach(results::index);
 
                 var indexMetadata = metadataFactory.fromRepo(snapshotName, index.getName());
                 indexMetadata.getAliases().fieldNames().forEachRemaining(alias -> {
                     var aliasResult = CreationResult.builder().name(alias);
-                    aliasResult.failureType(creationResult.getFailureType());
+                    aliasResult.failureType(creationResults.get(0).getFailureType());
```

Comment:
nit, this code might be quicker to read if it were tighter (it would be easy to compress w/out the log statement - which could be moved to outside the construction.